src/main/java/com/epam/aidial/cfg/service/hashing/HashCalculator.java

```diff
@@ -0,0 +1,37 @@
+package com.epam.aidial.cfg.service.hashing;
+
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import org.springframework.stereotype.Component;
+
+@Component
+public class HashCalculator {
+    private final ObjectWriter writer;
+
+    public HashCalculator(ObjectMapper mapper) {
+        this.writer = mapper.copy()
+        .registerModule(new JavaTimeModule())
+        .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
+        .enable(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY)
+        .enable(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS)
+        .enable(SerializationFeature.WRITE_BIGDECIMAL_AS_PLAIN)
+        .writer();
+    }
+
+    public String calculateHash(Object body) {
+        try {
+            byte[] json = writer.withoutAttribute("createAt")
```

Comment:
let's profile big import over full db. have some perf concerns

Reply:
@siarhei-fedziukovich, Currently, the method is not called on import. The hash is calculated in isolated cases when an update is performed via the controller.

Reply:
but it will be used. no actions now. just to remember